anomalib/data/utils/augmenter.py

```diff
@@ -147,9 +155,14 @@ def augment_batch(self, batch: Tensor) -> Tuple[Tensor, Tensor]:
         masks = torch.stack(masks_list).to(batch.device)
 
         # Apply perturbations batch wise
-        beta = torch.rand(batch_size) * 0.8
-        beta = beta.view(batch_size, 1, 1, 1).expand_as(batch).to(batch.device)
-
-        augmented_batch = batch * (1 - masks) + (1 - beta) * perturbations + beta * batch * (masks)
+        if isinstance(self.beta, float):
```

Comment:
a remark: i tend to do all sorts of validations and parameter normalization (in the sense dealing with multiple cases like this one) at the very beginning of `__init__`

it's kind of a matter of tast, but i think it's a sligthtly better practice to make the code structure more predictable

Reply:
Not sure if this will work in `__init__`. Say we select a batch size of 8 and it happens that the last batch is of size 7 or 1, then beta will have a wrong shape and the `augment_batch` computation might throw an error. While in most cases these checks and computations are redundant, it makes sense to keep them here for the edge cases.